lighthouse-core/gather/gather-runner.js

```diff
@@ -150,8 +150,19 @@ class GatherRunner {
     if (!mainRecord) {
       errorDef = LHError.errors.NO_DOCUMENT_REQUEST;
     } else if (mainRecord.failed) {
-      errorDef = {...LHError.errors.FAILED_DOCUMENT_REQUEST};
-      errorDef.message += ` ${mainRecord.localizedFailDescription}.`;
+      const netErr = mainRecord.localizedFailDescription;
+      // Match all resolution and DNS failures
+      // https://cs.chromium.org/chromium/src/net/base/net_error_list.h?rcl=cd62979b
+      if (
+        netErr.startsWith('net::ERR_NAME_NOT_RESOLVED') ||
+        netErr.startsWith('net::ERR_NAME_RESOLUTION_FAILED') ||
+        netErr.startsWith('net::ERR_DNS_')
+      ) {
+        errorDef = {...LHError.errors.DNS_FAILURE};
```

Comment:
if no string/message altering, no need to object spread to get a copy (e.g. see `NO_DOCUMENT_REQUEST` above)